packages/universal-hooks-testing-library/test/main.ts

```diff
@@ -26,11 +26,342 @@ describe("React", () => {
 });
 
 function runTests() {
+  describe("Async Utils", () => {
+    describe("waitForNextUpdate", () => {
+      test("default", async () => {
+        function hook() {
+          const [counter, setCounter] = useState(0);
+
+          setTimeout(() => setCounter((c = 0) => c + 1), 10);
+
+          return {
+            counter,
+            setCounter,
+          };
+        }
+
+        const { result, waitForNextUpdate } = renderHook(hook);
+
+        expect(result.current?.counter).toBe(0);
+        await waitForNextUpdate();
+        expect(result.current?.counter).toBe(1);
+        await waitForNextUpdate();
+        expect(result.current?.counter).toBe(2);
+      });
+
+      test("timeout expiry", async () => {
+        let error: Error | undefined = undefined;
+
+        function hook() {
+          const [counter, setCounter] = useState(0);
+
+          setTimeout(() => setCounter((c = 0) => c + 1), 10);
+
+          return {
+            counter,
+            setCounter,
+          };
+        }
+
+        const { result, waitForNextUpdate } = renderHook(hook);
+
+        expect(result.current?.counter).toBe(0);
+
+        await waitForNextUpdate();
+        expect(result.current?.counter).toBe(1);
+
+        try {
+          await waitForNextUpdate({ timeout: 1 });
+        } catch (e) {
+          error = e;
+        }
+        expect(error?.name).toMatch(/^(TimeoutError|Error)$/);
+        expect(result.current?.counter).toBe(1);
+
+        await waitForNextUpdate();
+        expect(result.current?.counter).toBe(2);
+      });
+
+      test("default timeout", async () => {
+        let error: Error | undefined = undefined;
+
+        function hook() {
+          const [counter, setCounter] = useState(0);
+
+          setTimeout(() => setCounter((c = 0) => c + 1), 2000);
+
+          return {
+            counter,
+            setCounter,
+          };
+        }
+
+        const { result, waitForNextUpdate } = renderHook(hook);
+
+        expect(result.current?.counter).toBe(0);
+
+        try {
+          await waitForNextUpdate();
+        } catch (e) {
+          error = e;
+        }
+        expect(error?.name).toMatch(/^(TimeoutError|Error)$/);
+        expect(result.current?.counter).toBe(0);
+      });
+    });
+
+    describe("waitFor", () => {
+      // test truthy, false, undefined, and error states
+      test("resolve when callback returns truthy or undefined", async () => {
+        function hook() {
+          const [counter, setCounter] = useState(0);
+
+          setTimeout(() => setCounter((c = 0) => c + 1), 10);
+
+          return {
+            counter,
+            setCounter,
+          };
+        }
```

Comment:
Same comment as earlier about copy-pasta

```suggestion
        function hook() {
          const [counter, setCounter] = useState(0);

          setTimeout(() => setCounter((c = 0) => c + 1), 10);

          return {
            counter,
            setCounter,
          };
        }
```

Reply:
Done.